crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -415,7 +429,7 @@ impl<'db> SemanticIndexBuilder<'db> {
         self.pop_scope();
         assert!(self.scope_stack.is_empty());
 
-        assert!(self.current_assignment.is_none());
+        assert_eq!(self.current_assignments.len(), 0);
```

Comment:
Nit: i like to do 
```suggestion
        assert_eq!(&self.current_assignments, &[]);
```

it gives better error messages if the vec isn't empty by printing all assignments.

Reply:
OK, changed to that (along with deriving `PartialEq` on `CurrentAssignment` to get that to work, turns out we hadn't been doing any equality checks on those before this change)